### DISCUSSION TITLE
Is it possible to use virtio-fs in windows guest?

### DISCUSSION THREAD

**Discussion by fabceolin** [vagrant-libvirt/vagrant-libvirt#1324] [Q&A]
Hi,

I would like to use vagrant-libvirt to use windows guests with virtio-fs enabled.
I setup a ubuntu 20.04 following the readme.md from this repository instructions (with this ppa enabled https://launchpad.net/~savoury1/+archive/ubuntu/virtualisation) with vagrant 2.2.17 and made a Vagrant file with this session:

```
Vagrant.configure("2") do |config|
  config.vm.provider :libvirt do |libvirt|
    libvirt.cpus = 2
    libvirt.memory = 8192
    libvirt.memorybacking :access, :mode => "shared"
  end
  config.vm.synced_folder "./", "/vagrant", type: "virtiofs"
end
```

The windows machine goes up, but the PCI devices related to virtio-fs didn't appear. 

How can I made a virtio-fs device appear in device manager?

Related links:

https://virtio-fs.gitlab.io/howto-windows.html there are instruction how to enable virtio-fs under Windows. 

Here there are some discusstion about the subject https://github.com/virtio-win/kvm-guest-drivers-windows/issues/473

Thanks

**Comment by electrofelix** [comment]
Sorry, I missed this question coming in a while back.

At the moment it's only linux guests that are supported:
https://github.com/vagrant-libvirt/vagrant-libvirt/blob/81b6fb715a661ecbd85f1b94177029ca52315b52/lib/vagrant-libvirt/plugin.rb#L39-L42

https://github.com/vagrant-libvirt/vagrant-libvirt/blob/master/lib/vagrant-libvirt/cap/mount_virtiofs.rb only covers handling this for linux guests.

It should be possible to add the same for Windows, but based on your comments your windows guest doesn't support virtio-fs, and you'd need to solve that which is outside of this projects scope.

Once that is solved it shouldn't be too difficult to add the additional capability to perform the mount automatically in the windows guest.

**Reply by fabceolin** [reply]
I would like to do tests like https://github.com/virtio-win/kvm-guest-drivers-windows/issues/781 using vagrant-libvirt. Is there anything that I can do to enable the virtiofs PCI Device on windows guests?

**Reply by electrofelix** [reply]
As I understand it, there is nothing stopping enabling of the device by setting the synced_folder type to virtiofs. See https://github.com/vagrant-libvirt/vagrant-libvirt/tree/81b6fb715a661ecbd85f1b94177029ca52315b52#synced-folders

You will need to enable some backing shared memory as it's not done automatically for you by vagrant-libvirt. This should make the device available and if you then install the device driver in Windows you should be able to mount it in the guest.

Presumably it should be possible to add code to vagrant-libvirt to install the device drivers automatically and mount the folder by adding a similar windows capability for the mount function.

**Comment by rgl** [comment]
I would also like to use virtio-fs with Windows. But, I'm currently stuck at https://gitlab.com/virtio-fs/virtiofsd/-/issues/57.

**Reply by electrofelix** [reply]
Looks like there has been some progress at least though, so possibly worth making it possible to enable for windows guests with an experimental warning.
